micro_sam/instance_segmentation.py

```diff
@@ -1139,3 +1143,145 @@ def get_amg(
         segmenter = TiledInstanceSegmentationWithDecoder(predictor, decoder, **kwargs) if is_tiled else\
             InstanceSegmentationWithDecoder(predictor, decoder, **kwargs)
     return segmenter
+
+
+def automatic_instance_segmentation(
+    input_path: Union[os.PathLike, str],
+    output_path: Union[os.PathLike, str],
+    embedding_path: Optional[Union[os.PathLike, str]] = None,
+    model_type: str = util._DEFAULT_MODEL,
+    checkpoint_path: Optional[Union[os.PathLike, str]] = None,
+    key: Optional[str] = None,
+    ndim: Optional[int] = None,
+    tile_shape: Optional[Tuple[int, int]] = None,
+    halo: Optional[Tuple[int, int]] = None,
+    use_amg: bool = False,
+    **generate_kwargs
+) -> None:
+    """Run automatic segmentation for the input image.
+
+    Args:
+        input_path: input_path: The input image file(s). Can either be a single image file (e.g. tif or png),
+            or a container file (e.g. hdf5 or zarr).
+        output_path: The output path where the instance segmentations will be saved.
+        embedding_path: The path where the embeddings are cached already / will be saved.
+        model_type: The SegmentAnything model to use. Will use the standard vit_l model by default.
+        checkpoint_path: Path to a checkpoint for a custom model.
+        key: The key to the input file. This is needed for container files (eg. hdf5 or zarr)
+            or to load several images as 3d volume. Provide a glob patterm, eg. "*.tif", for this case.
+        ndim: The dimensionality of the data.
+        tile_shape: Shape of the tiles for tiled prediction. By default prediction is run without tiling.
+        halo: Overlap of the tiles for tiled prediction.
+        use_amg: Whether to use Automatic Mask Generation (AMG) as the automatic segmentation method.
+    """
+    predictor, state = util.get_sam_model(model_type=model_type, checkpoint_path=checkpoint_path, return_state=True)
```

Comment:
We load the checkpoint twice if we do this here and in line 1178. It would be good to load only once to avoid overhead.

Reply:
Yes, true. This one should be updated now!